python/cucim/src/cucim/skimage/segmentation/_chan_vese.py

```diff
@@ -89,7 +89,7 @@ def _cv_calculate_variation(image, phi, mu, lambda1, lambda2, dt):
     similar approach is used by Rami Cohen, and it is from there that the
     C1-4 notation is taken.
     """
-    eta = 1e-16
+    eta = cp.asarray(1e-16, dtype=phi.dtype)
```

Comment:
Also do we need to apply this in other cases?

```suggestion
    eta = cp.asarray(1e-16, dtype=phi.dtype)
    mu = cp.asarray(mu, dtype=phi.dtype)
    lambda1 = cp.asarray(lambda1, dtype=phi.dtype)
    lambda2 = cp.asarray(lambda2, dtype=phi.dtype)
    dt = cp.asarray(dt, dtype=phi.dtype)
```

Reply:
except for `eta` those are done outside of the function call already. I should probably refactor so `eta` is also passed into the argument so this doesn't have to be converted repeatedly in the loop